src/main/java/carpet/helpers/EntityPlayerActionPack.java

```diff
@@ -238,7 +250,16 @@ public void onUpdate()
                 }
             }
         }
-        float vel = sneaking?0.3F:1.0F;
+
+        if (player.isUsingItem())
+        {
+            ItemStack item = player.getUseItem();
+            ((ServerPlayerInterface) player).getActionPack().setSlowed(item.isEdible() || item.is(Items.SHIELD) || item.is(Items.BRUSH) || item.is(Items.BOW) || item.is(Items.CROSSBOW) || item.is(Items.TRIDENT) || item.is(Items.GOAT_HORN) || item.is(Items.SPYGLASS));
```

Comment:
Isn't this equivalent to just `isUsingItem`, or am I missing any?

Also given nothing else is using `slowed`, it could be a local.

Haven't checked more of the code yet, sorry.

Reply:
Yep, looks like `player.isUsingItem()` is only true while using any of those listed items, most of that was completely unneeded. I'll commit a cleaned up version